json_schemas/sample_unit_tests.py

```diff
@@ -0,0 +1,205 @@
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
+
+  def test_memory_debugging(self):
+    config_name = 'complete_config_08_memory_debugging.json'
+    print('Testing {0}'.format(config_name))
+    with open(os.path.join('tutorial_configs', config_name), 'r') as infile:
+      config_json = json.load(infile)
+
+    try:
+      submitty_json_validator.submitty_validate_schemas(config_json,SCHEMA)
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
+
+  def test_java_testing(self):
+    config_name = 'complete_config_09_java_testing.json'
+    print('Testing {0}'.format(config_name))
+    with open(os.path.join('tutorial_configs', config_name), 'r') as infile:
+      config_json = json.load(infile)
+
+    try:
+      submitty_json_validator.submitty_validate_schemas(config_json,SCHEMA)
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
+
+  def test_java_coverage(self):
+    config_name = 'complete_config_10_java_coverage.json'
+    print('Testing {0}'.format(config_name))
+    with open(os.path.join('tutorial_configs', config_name), 'r') as infile:
+      config_json = json.load(infile)
+
+    try:
+      submitty_json_validator.submitty_validate_schemas(config_json,SCHEMA)
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
+
+  def test_resources(self):
+    config_name = 'complete_config_11_resources.json'
+    print('Testing {0}'.format(config_name))
+    with open(os.path.join('tutorial_configs', config_name), 'r') as infile:
+      config_json = json.load(infile)
+
+    try:
+      submitty_json_validator.submitty_validate_schemas(config_json,SCHEMA)
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
+
+  def test_system_calls(self):
+    config_name = 'complete_config_12_system_calls.json'
+    print('Testing {0}'.format(config_name))
+    with open(os.path.join('tutorial_configs', config_name), 'r') as infile:
+      config_json = json.load(infile)
+
+    try:
+      submitty_json_validator.submitty_validate_schemas(config_json,SCHEMA)
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
+
+  def test_cmake_compilation(self):
+    config_name = 'complete_config_13_cmake_compilation.json'
+    print('Testing {0}'.format(config_name))
+    with open(os.path.join('tutorial_configs', config_name), 'r') as infile:
+      config_json = json.load(infile)
+
+    try:
+      submitty_json_validator.submitty_validate_schemas(config_json,SCHEMA)
+      success = True
+    except Exception as e:
+      print(e)
+      success = False
+
+    self.assertTrue(success)
```

Comment:
You can write this much more succinctly using [unittest.subTest](https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest) (I eyeballed this code so may need some slight tweaking):
```
import unittest
import schema_validator as submitty_json_validator
import json
import os 

test_files = [
    'complete_config_01_simple_python',
    'complete_config_02_simple_cpp',
    'complete_config_03_multipart',
    'complete_config_04_python_static_analysis',
    'complete_config_05_cpp_static_analysis',
    'complete_config_06_loop_types',
    'complete_config_07_loop_depth,
    'complete_config_08_memory_debugging',
    'complete_config_09_java_testing',
    'complete_config_10_java_coverage',
    'complete_config_11_resources',
    'complete_config_12_system_calls',
    'complete_config_13_cmake_compilation'
]

class TestSchema(unittest.Testcase):
    def test_validate_schema(self):
        for test_file in test_files:
            with self.subTest(msg='Testing validation of test_file', test_file=test_file):
                with open(os.path.join('tutorial_configs', test_file), 'r') as infile:
                    config_json = json.load(infile)

                try:
                    submitty_json_validator.submitty_validate_schemas(config_json, SCHEMA)
                    success = True
                except Exception as e:
                    print(e)
                    success = False
                self.assertTrue(success)
```

Reply:
Thanks very much for the tip! I figured there must be a less verbose way to write these when there is so much overlap. I'll get the change in soon.